grafana/grafana
# loki: handle parsing infinity in loki response

Loki can return `+inf` and `-inf` as numeric values

Loki returns numeric values as strings-that-contain-a-float.

in loki-frontend-mode we use the javascript `parseFloat` function to go from string to number, and that does not know `+inf` and `-inf`.  (fun fact: it handles `+Infinity` and `-Infinity`, but that does not help us 😿 )

in prometheus-frontend-mode this was fixed in https://github.com/grafana/grafana/pull/28288 by using this function: https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/prometheus/result_transformer.ts#L582-L591  , so i used the same code here too (simplified it a little).

how to test:
- if the unit-test goes through things should be ok, but if you really want to try it manually, continue reading 😁 
- start a Loki db
- go to explore-mode
- you need some data in Loki from which we can `unwrap` a number.
- let's say the data is in JSON, and the number is in the field named "num"
- run a query like `quantile_over_time(42,{level="info"}|json num="num"|unwrap num [10s])` (the important part is the value `42`. any number larger than`1` works)
- open the Inspector/Data and verify that you get `Infinity` as value
- run a query like `quantile_over_time(42,{level="info"}|json num="num"|unwrap num [10s]) *  -1`)
- open the Inspector/Data and verify that you get `-Infinity` as value
